### PR TITLE
Change error message when failed to add friend because user not found

### PR DESCRIPTION
**Pull request by karenfrilya97**

### CODE REVIEW

File: src/client/components/AddFriendModal/AddFriendModal.jsx [hunk 1]
```diff
@@ -69,12 +69,15 @@ class AddFriendModal extends Component {
         window.location.reload();
       })
       .catch(function(error) {
-        if (error.response && error.response.status == 401) {
+        if (error.response && error.response.status === 401) {
           instance.routeChange(PATHS.landingPage);
```
Comment (tanrenyi) [inline src/client/components/AddFriendModal/AddFriendModal.jsx]: use `switch`

```js
.catch (function(error) {
  if (error.response) {
    switch (error.response.status) {
      case 401:
        instance.routeChange(PATHS.landingPage);
        return;
      case 404:
        alert(`Could not find user with email '${email}'.`);
      default:
        alert(error.message);
        console.log(error);
        return
    }
  }
  alert(error.message);
  console.log(error);
})
```

File: src/server/api/routes/trip.js [hunk 2]
```diff
@@ -151,15 +151,30 @@ router.get('/:tripId/members', function (req, res) {
...
-    const addTripFriend = tripFriendQueryModel.addTripFriend({ trip_id: req.params.tripId, email: req.body.email});
-
-    addTripFriend
+    const { tripId } = req.params;
+    const { email } = req.body;
+    const userNotFoundMessage = `Could not find user with email '${email}'`;
+
+    userQueryModel.getUserId({ email })
+        .then(function (results) {
+            if (results.length === 0) {
```
Comment (tanrenyi) [inline src/server/api/routes/trip.js]: does this work?
```js
if(!results) {

```

File: src/client/components/AddFriendModal/AddFriendModal.jsx [hunk 3]
```diff
@@ -69,12 +69,15 @@ class AddFriendModal extends Component {
         window.location.reload();
       })
       .catch(function(error) {
-        if (error.response && error.response.status == 401) {
+        if (error.response && error.response.status === 401) {
```
Comment (tanrenyi) [inline src/client/components/AddFriendModal/AddFriendModal.jsx]: use `switch`

```js
.catch (function(error) {
  if (error.response) {
    switch (error.response.status) {
      case 401:
        instance.routeChange(PATHS.landingPage);
        return;
      case 404:
        alert(`Could not find user with email '${email}'.`);
      default:
        alert(error.message);
        console.log(error);
        return
    }
  }
  alert(error.message);
  console.log(error);
})
```

File: src/server/api/routes/trip.js (hunk 2 above)
Comment (karenfrilya97) [reply src/server/api/routes/trip.js]: no, `![]` returns `false`.

File: src/client/components/AddFriendModal/AddFriendModal.jsx (hunk 3 above)
Comment (karenfrilya97) [reply src/client/components/AddFriendModal/AddFriendModal.jsx]: but then the default will be repeated right? i think i prefer to use if else.

**Pull request merged**